Add an operator overview diagram (#3065)
tenzir/vast: web/docs/understand/language/pipelines.md

If a pipeline would not have a source and sink, it would "leak" data. We call
pipelines that have both a source and sink a **closed pipeline**. VAST can only
execute closed pipelines. A pipeline that solely consists of a chain
transformations is an **open pipeline**.

## Operator Overview

Let's take an inside look at the anatomy of a pipeline: the operator building
blocks. The diagram below shows the three major customization points of the

> pipeline execution engine. The operator SDK consists of transformations,

Reviewer: ```suggestion
pipeline execution engine. The operator SDK allows for building transformations,
```

Technically speaking the "SDK" would allow for building further. It's not the set of operators we ship ourselves.

sources, and sinks. The SDK for [connectors and
formats](#connectors-and-formats) are convience abstractions to make it easy to
get data in and out of the system.[^1]

[^1]: It is also possible to write source and sink operators, but this requires
      a bit deeper understanding of system internals. In most of the cases,
      adding a data source or sink by means of a connector or format is

>       sufficient.

Reviewer: Rephrasing suggestion:

> `from <connector> read <format>` and `to <connector> write <format>` are just operators that any developer can write. They are a bit more complex than operators transforming events, so we provide the connector and format SDKs to make it easier for developers to add sources and sinks.

![Operator Overview](operator-overview.excalidraw.svg)

:::caution Work in Progress
The above diagram shows the scope of our minimum viable product at launch time
later this year. We are not fully there yet with our new pipeline execution
engine. Most of the code exists already, but needs to moved into this new
composable connector/format framework. Most of the transformations exist
already (only `sort` and `inspect` are missing), but we haven't ported majority
of formats and connectors yet. Until then, please consider this diagram for

> architectureal exposition only.

Author: I simplified this.
